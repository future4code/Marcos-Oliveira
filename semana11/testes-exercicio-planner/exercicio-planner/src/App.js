import React, { useState, useEffect } from 'react'
import axios from 'axios';
import { AppContainer, SectionContainer, Header, OptionsHeader, DaysSection } from './styles.js';

const App = () => {
  const [tasks, setTasks] = useState([])
  const [inputTask, setInputTask] = useState("")
  const [selectDay, setSelectDay] = useState("")
  let orderedTasks = {
                        monday: [],
                        tuesday: [],
                        wednesday: [],
                        thursday: [],
                        friday: [],
                        saturday: [],
                        sunday: []
                      }

  useEffect(() => {
    getTaskList()
    
  }, [tasks])

  const getTaskList = async() => {
    try {
      const response = await axios.get("https://us-central1-labenu-apis.cloudfunctions.net/generic/planner-mello-marcos-oliveira")
      
      setTasks(response.data)
    } catch (e) {
      alert("Erro ao buscar tarefas")
    }
  }

  for (let task of tasks) {
    switch (task.day) {
      case "segunda" :
        orderedTasks.monday.push(task)
        break;
      case "terca" :
        orderedTasks.tuesday.push(task)
        break;
      case "quarta" :
        orderedTasks.wednesday.push(task)
        break;
      case "quinta" :
        orderedTasks.thursday.push(task)
        break;
      case "sexta" :
        orderedTasks.friday.push(task)
        break;
      case "sabado" :
        orderedTasks.saturday.push(task)
        break;
      case "domingo" :
        orderedTasks.sunday.push(task)
        break;
      default:
        return
    }
  }

  const onChangeInput = (event) => {
    setInputTask(event.target.value)
  }

  const onChangeSelect = (event) => {
    setSelectDay(event.target.value)
  }

  const createTask = async() => {
    if (inputTask === "" || selectDay === "") {
      alert("Campo Nova tarefa ou dia da semana vazio")
    
    } else {

      const body = {
        text: inputTask,
        day: selectDay
      }

      try {
        await axios.post("https://us-central1-labenu-apis.cloudfunctions.net/generic/planner-mello-marcos-oliveira", body)

        getTaskList()
        setInputTask("")
      } catch (e) {
        alert("Erro ao criar tarefa")
      }
    }
  }

  const deleteTask = (id) => {
    axios.delete(`https://us-central1-labenu-apis.cloudfunctions.net/generic/planner-mello-marcos-oliveira/${id}`)
         .then(() => {
           
           getTaskList()
         })
         .catch(() => {
           alert("Não foi possivel excluir a tarefa")
         })
  }

  return (
    <AppContainer>
      <SectionContainer>
        <Header>
          <h1>MINHA SEMANA</h1>
            <OptionsHeader>
              <label htmlFor="tasks">Nova Tarefa: 
                <input 
                  htmlFor="tasks"
                  type="text"
                  value={inputTask}
                  onChange={onChangeInput}
                />
              </label>
              <label>Escolha o dia:
                <select
                  value={selectDay}
                  onChange={onChangeSelect}
                >
                  <option value=""></option>
                  <option value="segunda">Segunda-Feira</option>
                  <option value="terca">Terça-Feira</option>
                  <option value="quarta">Quarta-Feira</option>
                  <option value="quinta">Quinta-Feira</option>
                  <option value="sexta">Sexta-Feira</option>
                  <option value="sabado">Sábado</option>
                  <option value="domingo">Domingo</option>
                </select>
              </label>
              <button onClick={createTask}>Criar Tarefa</button>
            </OptionsHeader>
        </Header>
        <DaysSection>
          <article>
            <p>Segunda</p>
            <ul>
              {orderedTasks.monday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>
          <article>
            <p>Terça</p>
            <ul>
              {orderedTasks.tuesday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>
          <article>
            <p>Quarta</p>
            <ul>
              {orderedTasks.wednesday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>
          <article>
            <p>Quinta</p>
            <ul>
              {orderedTasks.thursday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>
          <article>
            <p>Sexta</p>
            <ul>
              {orderedTasks.friday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>
          <article>
            <p>Sábado</p>
            <ul>
              {orderedTasks.saturday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>
          <article>
            <p>Domingo</p>
            <ul>
              {orderedTasks.sunday.map((task) => {
                return <li key={task.id} onDoubleClick={() => deleteTask(task.id)}>{task.text}</li>
              })}
            </ul>
          </article>              
        </DaysSection>
      </SectionContainer>
      <p>Clique duplo apaga a tarefa</p>
    </AppContainer>
  );
}

export default App;
