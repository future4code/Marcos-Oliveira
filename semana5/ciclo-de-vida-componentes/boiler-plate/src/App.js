import React from 'react'
import styled from 'styled-components'
import './styles.css'

const TarefaList = styled.ul`
  padding: 0;
  width: 200px;
`

const Tarefa = styled.li`
  padding-top: 20px;
  text-align: left;
  text-decoration: ${({completa}) => (completa ? 'line-through' : 'none')};
`

const InputsContainer = styled.div`
  display: grid;
  grid-auto-flow: column;
  gap: 10px;
`

const BtnDeletar = styled.button `
  margin-right: 0px;
  font-size: 12px;
`
const BtnEditar = styled.button `
  font-size: 12px;
  margin-left: 10px;
`

class App extends React.Component {
    state = {
      tarefas: [],
      inputValue: "",
      filter: "",
      filterNome: ""
    }

  componentDidUpdate() {
    localStorage.setItem("listaDeTarefas", JSON.stringify(this.state.tarefas));
  };

  componentDidMount() {
    if (localStorage.getItem("listaDeTarefas")) {
      const listaDeTarefas = JSON.parse(localStorage.getItem("listaDeTarefas"));

      this.setState({tarefas: listaDeTarefas});
    }
  };

  onChangeInput = (event) => {
    this.setState({inputValue: event.target.value});
  }

  criaTarefa = () => {
    const tarefaNova = {
      id: Date.now(),
      texto: this.state.inputValue,
      completa: false
    };

    const listaNovaTarefa = [...this.state.tarefas, tarefaNova];
    this.setState({tarefas: listaNovaTarefa, inputValue: ""});
  }

  editaTarefa = (id) => {
    const tarefaEditada = this.state.tarefas.map((tarefa) => {
      if (id === tarefa.id) {
        const novaLista = {
          ...tarefa,
          texto: this.state.inputValue
        }
        return novaLista
      }
    })
    this.setState({tarefas: tarefaEditada, inputValue: ""});
  }

  apagaTarefa = () => {
    const apagaTodasTarefas = [];
    this.setState({tarefas: apagaTodasTarefas});
  }

  selectTarefa = (id) => {
    const tarefaCompleta = this.state.tarefas.map((tarefa) => {
      if (id === tarefa.id) {
        const novaLista = {
          ...tarefa,
          completa: !tarefa.completa
        }
        return novaLista
      } else {
        return tarefa
      }
    })
    this.setState({tarefas: tarefaCompleta});
  }

  removeTarefa = (id) => {
    const novaLista = this.state.tarefas.filter((tarefa) => {
      return id !== tarefa.id
    }) 

    this.setState({tarefas: novaLista});
  }

  ordenaDeAZ = () => {
    const listaDeAZ = this.state.tarefas
    listaDeAZ.sort(function (a, b) {
      return (a.texto > b.texto) ? 1 : ((b.texto > a.texto) ? -1 : 0);
    });
    this.setState({tarefas: listaDeAZ});
  }

  ordenaDeZA = () => {
    const listaDeZA = this.state.tarefas
    listaDeZA.sort(function (a, b) {
      return (a.texto > b.texto) ? 1 : ((b.texto > a.texto) ? -1 : 0);
    });
    listaDeZA.reverse();
    this.setState({tarefas: listaDeZA});
  }

  onChangeFilter = (event) => {
    this.setState({filter: event.target.value});
  }

  onChangeFilterNome = (event) => {
    this.setState({filterNome: event.target.value});
  }

  render() {
    const listaFiltrada = this.state.tarefas
      .filter(tarefa => {
        switch (this.state.filter) {
          case 'pendentes':
            return !tarefa.completa
          case 'completas':
            return tarefa.completa
          default:
            return true
        }
      });

    const listaPorNome = this.state.tarefas
      

    return (
      <div className="App">
        <h1>Lista de tarefas</h1>
        <InputsContainer>
          <label>Tarefa:</label>
          <input value={this.state.inputValue} onChange={this.onChangeInput}/>
          <button onClick={this.criaTarefa}>Adicionar</button>
          <button onClick={this.apagaTarefa}>Apagar Tarefas</button>
          <button onClick={this.ordenaDeAZ}>Ordenar de A-Z</button>
          <button onClick={this.ordenaDeZA}>Ordenar de Z-A</button>
        </InputsContainer>
        <br/>

        <InputsContainer>
          <label>Filtro:</label>
          <select value={this.state.filter} onChange={this.onChangeFilter}>
            <option value="">Nenhum</option>
            <option value="pendentes">Pendentes</option>
            <option value="completas">Completas</option>
          </select>
          <label>Filtro por Nome:</label>
          <select value={this.state.filterNome} onChange={this.onChangeFilterNome}>
            <option value="">Nenhum</option>
            {listaPorNome.map(tarefa => {
              return (
                <option>{tarefa.texto}</option>
              )
            })}
          </select>
        </InputsContainer>
        <TarefaList>
          {listaFiltrada.map(tarefa => {
            return (
              <div>
                <Tarefa
                  completa={tarefa.completa}
                  onClick={() => this.selectTarefa(tarefa.id)}
                >
                  {tarefa.texto} 
                </Tarefa>
                <BtnDeletar onClick={() => this.removeTarefa(tarefa.id)}>Deletar</BtnDeletar>
                <BtnEditar onClick={() => this.editaTarefa(tarefa.id)}>Editar</BtnEditar>
                <hr />
              </div>
            )
          })}
        </TarefaList>
      </div>
    )
  }
}

export default App
