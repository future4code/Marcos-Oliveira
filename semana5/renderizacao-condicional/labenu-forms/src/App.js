import React from 'react';
import styled from 'styled-components';
import Etapa1 from './componentes/Etapa1';
import Etapa2 from './componentes/Etapa2';
import Etapa3 from './componentes/Etapa3';
import Final from './componentes/Final';



class App extends React.Component {

  render() {

  

    return (
      <div className="App">
        
        <Final />
      
      </div>
    );
  }
}

export default App;
