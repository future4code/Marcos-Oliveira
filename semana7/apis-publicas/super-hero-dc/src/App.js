import React from 'react';
import Axios from 'axios';
import styled from 'styled-components';

const ContainerApp = styled.div `
  display: flex;
  flex-direction: column;
  align-items: center;
`

const SelectCapsula = styled.select `
  width: 150px;
  padding: 5px;
  font-weight: bold;
  font-size: 15px;
`

const ImgCapsula = styled.img `
  display: flex;
  flex-direction: column;
  margin: 20px 20px;
`

class App extends React.Component {
  
  state = {
    capsulas: [],
    imagesCapsulas: []
  }

  componentDidMount = () => {
    Axios.get("https://api.spacexdata.com/v3/dragons")
         .then(response => {
           this.setState({capsulas: response.data})
         })
         .catch(error => {
           alert(error)
         })
  }

  ImageDragon = (event) => {
    const dragonName = event.target.value
    if (dragonName === "Dragon 1") {
      Axios.get("https://api.spacexdata.com/v3/dragons/dragon1")
           .then(response => {
             this.setState({imagesCapsulas: response.data.flickr_images})
             const imagensQueAparecem = this.state.imagesCapsulas.slice(2, 4)
             this.setState({imagesCapsulas: imagensQueAparecem})
           })
           .catch(err => {
             alert(err)
           })
    } else if (dragonName === "Dragon 2") {
      Axios.get("https://api.spacexdata.com/v3/dragons/dragon2")
           .then(response => {
             this.setState({imagesCapsulas: response.data.flickr_images})
           })
           .catch(err => {
             alert(err)
           })
    }
  }

  render() {

    const images = this.state.imagesCapsulas  ? (this.state.imagesCapsulas.map(image => {
      return <ImgCapsula src={image} alt="Imagem da Cápsula" />
    })
    ) : (
      <div />
    )
  
    return (
      <ContainerApp>
        <h1>Cápsulas Dragon SpaceX</h1>
        <SelectCapsula onChange={this.ImageDragon}>
          <option value=""></option>
          {this.state.capsulas.map(capsula => {
            return <option value={capsula.name}>{capsula.name}</option>
          })}
        </SelectCapsula>
        <div>{images}</div>
      </ContainerApp>
    );
  }
}

export default App;