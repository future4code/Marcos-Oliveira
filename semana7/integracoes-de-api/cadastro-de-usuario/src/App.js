import React from 'react';
import CadastroDeUsuarios from "./componentes/CadastroDeUsuarios";
import ListaDeUsuarios from "./componentes/ListaDeUsuarios";

class App extends React.Component {

  state = {
    telaCadastro: true
  };

  mudaTela = () => {
    this.setState({telaCadastro: !this.state.telaCadastro});
  };
 
  render() {
    if (this.state.telaCadastro) {
      return (
        <div className="App">
          <button onClick={this.mudaTela}>Ir para Lista Usu√°rios</button>
          <CadastroDeUsuarios />
        </div>
      )
    } else {
      return (
        <div className="App">
          <button onClick={this.mudaTela}>Voltar para Tela Cadastro</button>
          <ListaDeUsuarios />
        </div>
      )
    }
  }
}

export default App;
