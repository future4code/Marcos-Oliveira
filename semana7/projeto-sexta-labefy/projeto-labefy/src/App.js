import React from 'react';
import Axios from 'axios';
import styled from 'styled-components';

class App extends React.Component {

  state = {
    playlists: [],
    name: ""
  }

  onChangePlaylist = (event) => {
    this.setState({novaPlaylist: event.target.value})
  }

  criarPlaylist = () => {
    const body = {
      name: this.state.novaPlaylist
    }
    Axios.post("https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists",body,
      {
        headers: {
          Authorization: "marcos-oliveira-mello"
        }
      }
    ).then(() => {
      this.setState({novaPlaylist: ""})
    }).catch((error) => {
      alert(error)
    })
  }

  async mostraUsuario() {
    try {
      const response = await Axios.get("https://us-central1-labenu-apis.cloudfunctions.net/labefy/playlists",
      {
        headers: {
          Authorization: "marcos-oliveira-mello"
        }
      }) 
  
      this.setState({playlists: response.data.results.list})
    } catch(erro) {
      alert(erro.response);
    }
  };

  render() {

    
  
    return (
      <div className="App">
        <h1>Labefy</h1>
        <span>Nova Playlist: </span>
        <input
          value={this.state.novaPlaylist}
          onChange={this.onChangePlaylist}
        />
        <button onClick={this.criarPlaylist}>Criar Playlist</button>
        {this.state.playlists.map(playlist => {
          return <li>{playlist.name}</li>
        })}
      </div>
    );
  }
}

export default App;
