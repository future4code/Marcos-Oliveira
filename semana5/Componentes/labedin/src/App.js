import React from 'react';
import './App.css';
import foto from './Imagens/Image.jpeg';
import manutencao from './Imagens/computadores.png';
import onbyte from './Imagens/onbyte.png';
import email from './Imagens/email.png';
import endereco from './Imagens/endereco.png';
import CardGrande from './components/CardGrande/CardGrande';
import ImagemButton from './components/ImagemButton/ImagemButton';
import CardPequeno from './components/CardPequeno/CardPequeno';

function App() {
  return (
    <div className="App">
      <div className="page-section-container">
        <h2>Dados pessoais</h2>
        <CardGrande 
          imagem={foto} 
          nome="Marcos Roberto de Oliveira" 
          descricao="Sou de Taquaritinga, interior de São Paulo, trabalhei por um bom tempo com hardware mas agora quero me aprofundar na área de programação."
        />
        
        <ImagemButton 
          imagem="https://image.flaticon.com/icons/png/512/117/117472.png" 
          texto="Ver mais"
        />
      </div>

      <div className="page-section-container">
        <CardPequeno 
          imagem={email}
          span="Email: "
          descricao="marcosoliveira@outlook.com" 
        />
        
        <CardPequeno 
          imagem={endereco}
          span="Endereço: "
          descricao="Rua Santos Dumont" 
        />
      </div>

      <div className="page-section-container">
        <h2>Experiências profissionais</h2>
        <CardGrande 
          imagem={manutencao} 
          nome="Prodata" 
          descricao="Montagem e Manutenção de Micros e Notebooks." 
        />
        
        <CardGrande 
          imagem={onbyte} 
          nome="Onbyte" 
          descricao="Instrutor de cursos interativos." 
        />
      </div>

      <div className="page-section-container">
        <h2>Contato</h2>
        <CardPequeno  
          endereco="Rua Santos Dumont" 
          email="marcosoliveira@outlook.com" 
        />
      </div>

      <div className="page-section-container">
        <h2>Minhas redes sociais</h2>
        <ImagemButton 
          imagem="https://d2v9ipibika81v.cloudfront.net/uploads/sites/261/2017/01/facebook-logo-3.png" 
          texto="Facebook" 
        />        

        <ImagemButton 
          imagem="https://logodownload.org/wp-content/uploads/2014/09/twitter-logo-1-1.png" 
          texto="Twitter" 
        />        
      </div>
    </div>
  );
}

export default App;
