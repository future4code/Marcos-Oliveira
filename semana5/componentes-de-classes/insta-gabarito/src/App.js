import React from 'react';
import './App.css';
import Post from './components/Post/Post';
import styled from 'styled-components';

const CaixaDeTexto = styled.input `
  display: flex;
  justify-content: center;
  padding: 4px;
  width: 290px;
  margin-top: 6px;
`

const BotaoPost = styled.button `
  padding: 5px;
  margin: 6px 6px;
`

class App extends React.Component {

  state = {
    posts: [
      {
        nomeUsuario:"Paulinha",
        fotoUsuario: "https://picsum.photos/50/50",
        fotoPost: "https://picsum.photos/200/150"
      },
      {
        nomeUsuario:"Marcos",
        fotoUsuario: "https://picsum.photos/50/51",
        fotoPost: "https://picsum.photos/200/151"
      },
      {
        nomeUsuario:"Amanda",
        fotoUsuario: "https://picsum.photos/50/52",
        fotoPost: "https://picsum.photos/200/152"
      }
    ]
  }

  adicionaPost = () => {
    const novoPost = {
      
      novoUsuario: this.state.InputNomeUsuario,
      novoFoto: this.state.InputFotoUsuario,
      novoPosts: this.state.InputFotoPost
    
    }

    const novaPost = [...this.state.posts, novoPost]

    this.setState({posts: novaPost})

  }

  onChangeInputNomeUsuario = (event) => {
    this.setState({ InputNomeUsuario: event.target.value })
  }

  onChangeInputFotoUsuario = (event) => {
    this.setState({ InputFotoUsuario: event.target.value })
  }

  onChangeInputFotoPost = (event) => {
    this.setState({ InputFotoPost: event.target.value })
  }

  render() {

    const listaDePosts = this.state.posts.map((post) => {
      
      return (
        <Post
          nomeUsuario={post.nomeUsuario}
          fotoUsuario={post.fotoUsuario}
          fotoPost={post.fotoPost}
        />
      )
    })
  

    return (
      <div className={'app-container'}>
        <div>
          <CaixaDeTexto
            value={this.state.InputNomeUsuario}
            onChange={this.onChangeInputNomeUsuario}
            placeholder={"Nome do Usuário"}
          />

          <CaixaDeTexto
            value={this.state.InputFotoUsuario}
            onChange={this.onChangeInputFotoUsuario}
            placeholder={"URL da Foto do Usuário"}
          />

          <CaixaDeTexto
            value={this.state.InputFotoPost}
            onChange={this.onChangeInputFotoPost}
            placeholder={"URL da Foto do Post"}
          />
        </div>

        <BotaoPost onClick={this.adicionaPost}>Adicionar Post </BotaoPost>

        {listaDePosts}

      </div>
    );
  }
}

export default App;
