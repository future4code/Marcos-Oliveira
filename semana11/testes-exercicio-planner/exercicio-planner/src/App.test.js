import React from 'react';
import { render, wait } from '@testing-library/react';
import axios from 'axios';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect'
import App from './App';

describe("Renderização inicial", () => {
  test("Renderiza tudo corretamente", async() => {
    axios.get = jest.fn().mockResolvedValue({
      data: [
        {
          "id": "sYLgZCjyXTwr3G9lqNIP",
          "text": "Tarefa requisição teste",
          "day": "sabado",
        }
      ]
    })
    
    const { getByTestId, findByText } = render(<App />)

    const input = getByTestId("inputTasks")
    expect(input).toBeInTheDocument()
        
    expect(getByTestId("selectTasks")).toBeInTheDocument()
    expect(getByTestId("btnTask")).toBeInTheDocument()
    expect(getByTestId("days")).toBeInTheDocument()
    
    const task = await findByText("Tarefa requisição teste")
    expect(task).toBeInTheDocument()

    expect(axios.get).toHaveBeenCalled() 
    
  })
})

describe("Criar uma tarefa", () => {
  test("Cria uma tarefa com sucesso", async () => {
    axios.get = jest.fn().mockResolvedValue({
      data: [
        {
          "id": "sYLgZCjyXTwr3G9lqNIP",
          "texto": "tarefa teste",
          "dia": "segunda"
        }
      ]
    })

    const { getByTestId, getByText } = render(<App />)

    const input = getByTestId("inputTasks")
    await userEvent.type(input, "tarefa teste")
    expect(input).toHaveValue("tarefa teste")

    const select = getByTestId("selectTasks")
    userEvent.selectOptions(select, getByText(/Segunda-Feira/i))
    expect(getByText(/Segunda-Feira/i)).toBeInTheDocument()

    const button = getByTestId("btnTask")
    userEvent.click(button)

    await wait(() => expect(axios.get).toHaveBeenCalledTimes(1))
  })
})


