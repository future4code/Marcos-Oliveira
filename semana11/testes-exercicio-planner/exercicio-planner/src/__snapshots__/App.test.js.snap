// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização inicial Renderiza tudo corretamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-AxjAm bSjxOM"
      >
        <section
          class="sc-AxirZ iqyMvK"
        >
          <div
            class="sc-AxiKw kZUgMc"
          >
            <h1>
              MINHA SEMANA
            </h1>
            <div
              class="sc-AxhCb dFmymE"
            >
              <label>
                Nova Tarefa:
                <input
                  data-testid="inputTasks"
                  type="text"
                  value=""
                />
              </label>
              <label>
                Escolha o dia:
                <select
                  data-testid="selectTasks"
                >
                  <option
                    value=""
                  />
                  <option
                    value="segunda"
                  >
                    Segunda-Feira
                  </option>
                  <option
                    value="terca"
                  >
                    Terça-Feira
                  </option>
                  <option
                    value="quarta"
                  >
                    Quarta-Feira
                  </option>
                  <option
                    value="quinta"
                  >
                    Quinta-Feira
                  </option>
                  <option
                    value="sexta"
                  >
                    Sexta-Feira
                  </option>
                  <option
                    value="sabado"
                  >
                    Sábado
                  </option>
                  <option
                    value="domingo"
                  >
                    Domingo
                  </option>
                </select>
              </label>
              <button
                data-testid="btnTask"
              >
                Criar Tarefa
              </button>
            </div>
          </div>
          <div
            class="sc-AxhUy ejHdMV"
          >
            <article>
              <p>
                Segunda
              </p>
              <ul />
            </article>
            <article>
              <p>
                Terça
              </p>
              <ul />
            </article>
            <article>
              <p>
                Quarta
              </p>
              <ul />
            </article>
            <article>
              <p>
                Quinta
              </p>
              <ul />
            </article>
            <article>
              <p>
                Sexta
              </p>
              <ul />
            </article>
            <article>
              <p>
                Sábado
              </p>
              <ul />
            </article>
            <article>
              <p>
                Domingo
              </p>
              <ul />
            </article>
          </div>
        </section>
        <p>
          Clique duplo apaga a tarefa
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="sc-AxjAm bSjxOM"
    >
      <section
        class="sc-AxirZ iqyMvK"
      >
        <div
          class="sc-AxiKw kZUgMc"
        >
          <h1>
            MINHA SEMANA
          </h1>
          <div
            class="sc-AxhCb dFmymE"
          >
            <label>
              Nova Tarefa:
              <input
                data-testid="inputTasks"
                type="text"
                value=""
              />
            </label>
            <label>
              Escolha o dia:
              <select
                data-testid="selectTasks"
              >
                <option
                  value=""
                />
                <option
                  value="segunda"
                >
                  Segunda-Feira
                </option>
                <option
                  value="terca"
                >
                  Terça-Feira
                </option>
                <option
                  value="quarta"
                >
                  Quarta-Feira
                </option>
                <option
                  value="quinta"
                >
                  Quinta-Feira
                </option>
                <option
                  value="sexta"
                >
                  Sexta-Feira
                </option>
                <option
                  value="sabado"
                >
                  Sábado
                </option>
                <option
                  value="domingo"
                >
                  Domingo
                </option>
              </select>
            </label>
            <button
              data-testid="btnTask"
            >
              Criar Tarefa
            </button>
          </div>
        </div>
        <div
          class="sc-AxhUy ejHdMV"
        >
          <article>
            <p>
              Segunda
            </p>
            <ul />
          </article>
          <article>
            <p>
              Terça
            </p>
            <ul />
          </article>
          <article>
            <p>
              Quarta
            </p>
            <ul />
          </article>
          <article>
            <p>
              Quinta
            </p>
            <ul />
          </article>
          <article>
            <p>
              Sexta
            </p>
            <ul />
          </article>
          <article>
            <p>
              Sábado
            </p>
            <ul />
          </article>
          <article>
            <p>
              Domingo
            </p>
            <ul />
          </article>
        </div>
      </section>
      <p>
        Clique duplo apaga a tarefa
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
